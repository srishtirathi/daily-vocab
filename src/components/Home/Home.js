import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Home.module.css';

const Home = () => (
  <div className={styles.home}>
    <h1>Daily Vocab!</h1>
    <h3>Learn a new word everyday</h3>
    <Link to="/allwords">
      {' '}
      <button type="button">Get Started</button>
    </Link>
  </div>
);

export default Home;
