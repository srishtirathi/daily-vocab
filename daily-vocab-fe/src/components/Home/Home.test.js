import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import {
  BrowserRouter,
} from 'react-router-dom';
import Home from './Home';

describe(Home.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('sanity check', () => {
    expect(1 + 1).toBe(2);
  });
  it('should match snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        {' '}
        <Home />
        {' '}
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should display intro message', () => {
    render(
      <BrowserRouter>
        {' '}
        <Home />
        {' '}
      </BrowserRouter>,
    );
    screen.getByText('Daily Vocab!');
    screen.getByText('Learn a new word everyday');
  });
  test('Should take to All Words on click', () => {
    render(<BrowserRouter><Home /></BrowserRouter>);
    const allWordsLink = screen.getByText('Get Started');
    fireEvent.click(allWordsLink);
    expect(document.location.href).toBe('http://localhost/allwords');
  });
});
