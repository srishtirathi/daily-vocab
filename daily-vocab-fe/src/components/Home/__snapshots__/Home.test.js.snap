// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
   
  <div
    class="home"
  >
    <h1>
      Daily Vocab!
    </h1>
    <h3>
      Learn a new word everyday
    </h3>
    <a
      href="/allwords"
    >
       
      <button
        type="button"
      >
        Get Started
      </button>
    </a>
  </div>
   
</div>
`;
